### PR TITLE
Feature/cid to string from string

### PR DESCRIPTION
**Pull request by artyom-yurin**
Added toString and fromString for CID codec
Also added base32

### CODE REVIEW

File: include/libp2p/multi/content_identifier_codec.hpp [hunk 1]
```diff
@@ -42,16 +43,23 @@ namespace libp2p::multi {
...
+     * @brief Encode CID to string representation
      * @param cid - input CID for encode
      * @param encoding - type of the encoding, ignored for CID v0 (always
      * Base58)
-     * @todo Sergey Kaprovich: FIL-133 add support for CID v1
      * @return CID string
      */
     static outcome::result<std::string> toString(
         const ContentIdentifier &cid,
         MultibaseCodec::Encoding encoding = MultibaseCodec::Encoding::BASE58);
```
Comment (turuslan) [inline include/libp2p/multi/content_identifier_codec.hpp]: Change default to BASE32_LOWER.
https://github.com/ipfs/go-cid/blob/master/cid.go#L347

File: src/multi/multibase_codec/codecs/base32.cpp [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+      return byte >> offset;
+    else
+      return byte << -offset;
+  }
+
+  unsigned char shift_left(uint8_t byte, int8_t offset) {
+    return shift_right(byte, -offset);
+  }
+
+  int encode_sequence(const uint8_t *plain, int len, char *coded,
```
Comment (turuslan) [inline src/multi/multibase_codec/codecs/base32.cpp]: use gls::span and/or std::string_view

File: include/libp2p/multi/content_identifier_codec.hpp (hunk 1 above)
Comment (sergkaprovich) [reply include/libp2p/multi/content_identifier_codec.hpp]: Both CID versions (0 and 1) can be encoded with Base58, and if multibase codec version is not explicitly specified - we assume, that this is CID v0, which can be encoded only with Base58. 
We should change default to BASE32_LOWER here if we plan to use it only with CIDv1

Comment (turuslan) [reply include/libp2p/multi/content_identifier_codec.hpp]: Maybe we can add "auto" enum value, or separate method that accepts "base" param, and let "toString()" use default BASE32_LOWER.

**Pull request merged**